src/components/UploadSong.tsx

```diff
@@ -42,18 +43,42 @@ interface SongProgressOverlayProps {
 const UploadSong: FunctionComponent<UploadSongProps> = ({
   file,
   onChange,
-  onError,
   onBlur,
   errorMessage,
+  isValidationTriggered,
+  resetValidationTrigger,
 }) => {
   const theme = useTheme();
 
   const [isHovering, setIsHovering] = useState(false);
   const [song, setSong] = useState<Howl | null>(null);
   const [songProgress, setSongProgress] = useState<number>(0);
   const [isSongPlaying, setIsSongPlaying] = useState<boolean>(false);
+  const [customErrorMessage, setCustomErrorMessage] = useState<string>("");
   const visualizerRef = useRef<HTMLCanvasElement>(null);
 
+  const AUDIO_MIN_DURATION_SEC = 60;
+
+  const AUDIO_MIN_FILE_SIZE_MB = 1;
+  const AUDIO_MAX_FILE_SIZE_GB = 1;
+
+  const createAudioBuffer = async (value: File) => {
+    const audioContext = new (window.AudioContext || window.AudioContext)();
+    const arrayBuffer = await value.arrayBuffer();
+
+    return audioContext.decodeAudioData(arrayBuffer);
+  };
+
+  const isFileSizeValid = (value: File) => {
+    const fileSizeInMB = value.size / (1024 * 1024);
+    const fileSizeInGB = value.size / (1024 * 1024 * 1024);
```

Comment:
Thinking about switching this and album max value to 1000 instead of 1024 factor, as files are typically in bytes and Megabytes/Gigabytes (MB/GB) not Mebibytes/Gebibytes (MiB/GiB). Do others agree?